packages/drivers/odsp-driver/src/createFile.ts

```diff
@@ -57,7 +59,7 @@ const isInvalidFileName = (fileName: string): boolean => {
  */
 export async function createNewFluidFile(
     getStorageToken: InstrumentedStorageTokenFetcher,
-    newFileInfo: INewFileInfo,
+    fileInfo: INewFileInfo | IExistingFileInfo,
```

Comment:
rather than overloading existing functions, which i don't think necessarily sense given the differing requirements. and wil their naming, i think we should make new functions for the new code, and extract shared code into un-exported functions that both can call

Reply:
this will also forces changes where this is called, which i think we want, as currently i don't think we'll have good telemetry around this new path

Reply:
Somewhat similar - the fact that there are 6 places with if(isNewFileInfo()) checks suggests code should be split between those two paths.

Reply:
Thanks for the feedback, I've now split the two functions and refactored their shared code to createNewUtil.ts